File: src/ConsentStatus.ts
export enum ConsentStatus {
  /**
   * Consent status is unknown. Ump.requestInfoUpdate needs to be called to update the ConsentStatus
   */
  UNKNOWN = 'UNKNOWN',

  /**
   * User consent not required.
   */
  NOT_REQUIRED = 'NOT_REQUIRED',

  /**
   * User consent obtained. Personalized vs non-personalized undefined.
   */
  OBTAINED = 'OBTAINED',

  /**
   * User consent required but not yet obtained.
   */
  REQUIRED = 'REQUIRED',
}

File: ios/ConsentStatus.swift
import ExpoModulesCore
import UserMessagingPlatform

// Source: https://developers.google.com/admob/ios/privacy/api/reference/Enums/UMPConsentStatus

enum ConsentStatus: String, Enumerable {
    /**
     * Unknown consent status
     */
    case unknown = "UNKNOWN"

    /**
     * User consent required but not yet obtained
     */
    case required = "REQUIRED"

    /**
     * Consent not required.
     */
    case notRequired = "NOT_REQUIRED"

    /**
     * Undocumented.
     */
    case obtained = "OBTAINED"

    static func fromUMPConsentStatus(consentStatus: UMPConsentStatus) -> ConsentStatus {
        switch consentStatus {
        case .required:
            return .required
        case .notRequired:
            return .notRequired
        case .obtained:
            return .obtained
        case .unknown:
            return .unknown
        default:
            return .unknown
        }
    }
}
